x-pack/plugins/security_solution/public/siem_migrations/rules/components/rule_details_flyout/index.tsx

```diff
@@ -21,16 +21,21 @@ import {
   EuiFlexGroup,
   EuiFlexItem,
   useGeneratedHtmlId,
+  EuiSkeletonLoading,
+  EuiSkeletonTitle,
+  EuiSkeletonText,
 } from '@elastic/eui';
 import type { EuiTabbedContentTab, EuiTabbedContentProps, EuiFlyoutProps } from '@elastic/eui';
 
 import type { RuleMigration } from '../../../../../common/siem_migrations/model/rule_migration.gen';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
 import {
   RuleOverviewTab,
   useOverviewTabSections,
 } from '../../../../detection_engine/rule_management/components/rule_details/rule_overview_tab';
 import type { RuleResponse } from '../../../../../common/api/detection_engine/model/rule_schema';
 
+import * as logicI18n from '../../logic/translations';
```

Comment:
what does _logic_ mean? I see they are API related translations

Reply:
I use logic to keep the wrapping hooks for API calls. Those hooks are responsible for any configurations and error handling of API calls. Like requests caching setup and showing error toast etc. This structure is used within detection & engine, thus I inherited it.

What is the preferred structure in Threat Hunting? If you think we can simplify things, by moving hooks closer to the APIs (inside api folder), we can adjust folders setup as well. I don't have a strong NO against that 😄

Reply:
I like the idea of moving them within the API directory, they are hooks to interact with the API in the end. They don't have much _logic_ inside 😄 
Not need to do it now though, I plan to do a code consolidation refactor later.

Reply:
Yes indeed! I also like the `SiemRulesMigrationsService` and hooks laying next to it, so we could move my hooks there as well to have same approach